numpyro/primitives.py

```diff
@@ -232,7 +233,17 @@ def process_message(self, msg):
         cond_indep_stack = msg['cond_indep_stack']
         frame = CondIndepStackFrame(self.name, self.dim, self.subsample_size)
         cond_indep_stack.append(frame)
-        batch_shape = self._get_batch_shape(cond_indep_stack)
+        expected_shape = self._get_batch_shape(cond_indep_stack)
+        dist_batch_shape = msg['fn'].batch_shape if msg['type'] == 'sample' else ()
+        overlap_idx = len(expected_shape) - len(dist_batch_shape)
+        if overlap_idx < 0:
+            raise ValueError('Expected dimensions within plate = {}, which is less than the '
+                             'distribution\'s batch shape = {}.'.format(len(expected_shape), len(dist_batch_shape)))
+        trailing_shape = expected_shape[overlap_idx:]
+        if sum(map(sub, lax.broadcast_shapes(trailing_shape, dist_batch_shape), dist_batch_shape)) > 0:
```

Comment:
nit: can we just compare `lax.broadcast_shapes(trailing_shape, dist_batch_shape)` and `dist_batch_shape`?

Reply:
Hmm, this seems related to https://github.com/pyro-ppl/numpyro/issues/355#issuecomment-535246983: to deal with the case `trailing_shape=(2, 1, 5)`, `dist_batch_shape=(1, 3, 5)`. I think that it is better to add a note here to reflect your idea.